### ISSUE TITLE
Avoid hardcode in installer paths

### ISSUE THREAD

**Issue by roman-khazanskii** [oleg-shilo/wixsharp#1152]
All of the samples that I looked into use simple hardcoded paths for files and directories like:

```
var project = new ManagedProject("MyProduct",
                      new Dir(@"C:\My Company\My Product",
                          new File("readme.txt")));
```

I added setup project to solution with installee project, so I would like to avoid hardcoded values and somehow get them from solution or by reflection. 

I managed to read assembly info with `System.Diagnostics.FileVersionInfo.GetVersionInfo(@"..\MyDll\bin\Release\my_dll.dll");` but that  looks kinda ugly and also requires hardcoded path.

Is it possible to achieve this in more correct manner?

**Comment by oleg-shilo** [comment]
Not sure I follow.
None of the sample projects is using an absolute path as you specified.

This is a typical sample definition:
```C#
new Project("MyProduct",
    new Dir(@"%ProgramFiles%\My Company\My Product",
        new File(@"Files\Bin\MyApp.exe",
```

All paths are relative. 

Hard-coding is a default WiX/WixSharp definition way but you can use wild cards if you want: https://github.com/oleg-shilo/wixsharp/wiki/Deployment-scenarios#including-files-dynamically-working-with-wild-card-source-location

**Comment by roman-khazanskii** [comment]
>Not sure I follow.

Sorry for me not being clear again.
I meant that path in "File" or "Files" is hardcoded, i.e. it's always assumed that files to be installed are in the same directory as setup project or that we know their filenames beforehand.

I, on the other hand, want to install whatever was build by another project in current solution, provide only project (or assembly) name and extract everything else from it. If that's possible of course.

**Comment by oleg-shilo** [comment]
Then you need to look at the sample I provided (the link).

You can include the content of any directory you like and even apply the file name pattern:

```C#
var project =
    new Project("MyProduct",
        new Dir(@"%ProgramFiles%\My Company\My Product",
            new Files(@"Release\*.*")
            . . .
```

**Comment by roman-khazanskii** [comment]
>You can include the content of any directory you like and even apply the file name pattern:

Yes, I understand, however I still need to spell "Release" (or, more specifically, `..\Release`). I was wondering if it's possible to get this path from solution or assembly info or something like that.

**Comment by roman-khazanskii** [comment]
Won't this be a path to setup project application?

**Comment by roman-khazanskii** [comment]
Fair enough. Thanks!
